File: app/src/main/java/igrek/songbook/songselection/tree/ScrollPosBuffer.java
package igrek.songbook.songselection.tree;

import java.util.HashMap;

import igrek.songbook.dagger.DaggerIoc;
import igrek.songbook.persistence.songsdb.SongCategory;
import igrek.songbook.songselection.ListScrollPosition;

public class ScrollPosBuffer {
	
	private HashMap<SongCategory, ListScrollPosition> storedScrollPositions = new HashMap<>();
	
	public ScrollPosBuffer() {
		DaggerIoc.getFactoryComponent().inject(this);
	}
	
	public void storeScrollPosition(SongCategory category, ListScrollPosition y) {
		if (y != null)
			storedScrollPositions.put(category, y);
	}
	
	public ListScrollPosition restoreScrollPosition(SongCategory category) {
		return storedScrollPositions.get(category);
	}
	
	public boolean hasScrollPositionStored(SongCategory category) {
		return storedScrollPositions.containsKey(category);
	}
}

File: app/src/main/java/igrek/songbook/songselection/tree/ScrollPosBuffer.kt
package igrek.songbook.songselection.tree

import igrek.songbook.dagger.DaggerIoc
import igrek.songbook.persistence.general.model.Category
import igrek.songbook.songselection.listview.ListScrollPosition

class ScrollPosBuffer {

    private val storedScrollPositions = hashMapOf<Category?, ListScrollPosition>()

    init {
        DaggerIoc.factoryComponent.inject(this)
    }

    fun storeScrollPosition(category: Category?, y: ListScrollPosition?) {
        if (y != null)
            storedScrollPositions[category] = y
    }

    fun restoreScrollPosition(category: Category?): ListScrollPosition? {
        return storedScrollPositions[category]
    }

    fun hasScrollPositionStored(category: Category?): Boolean {
        return storedScrollPositions.containsKey(category)
    }
}
